### PR TITLE
Handle nullable types and empty partitions before Dask-ML predict

### PR DESCRIPTION
**Pull request by sarahyurick**
Tagging @randerzander and @VibhuJawa 

Changes in `create_model.py` handle nullable types, such as for this example:

```
df = pd.DataFrame({
    "rough_day_of_year": pd.Series([0, 1, 2, 3], dtype='Int32'),
    "prev_day_inches_rained": pd.Series([0, 1, 2, 3], dtype='float32'),
    "rained": pd.Series([False, False, False, True])
})
c.create_table("train_set", df)

model_class = ".linear_model.LogisticRegression'"
if GPU:
    model_class = "'cuml" + model_class
else:
    model_class = "'sklearn" + model_class

c.sql(f"""
CREATE OR REPLACE MODEL model WITH (
    model_class = {model_class},
     wrap_predict = True,
     wrap_fit = False,
    target_column = 'rained'
) AS (
    SELECT *
    FROM train_set
)
""")

c.sql("""
SELECT * FROM PREDICT(
  MODEL model,
  SELECT * FROM train_set
)
""").compute()
```

Changes in `predict.py` handle empty partitions, modeled based on this Dask-ML PR: https://github.com/dask/dask-ml/pull/912

### CODE REVIEW

File: dask_sql/physical/rel/custom/create_model.py [hunk 1]
```diff
@@ -183,7 +184,13 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
 
             delayed_model = [delayed(model.fit)(x_p, y_p) for x_p, y_p in zip(X_d, y_d)]
             model = delayed_model[0].compute()
-            model = ParallelPostFit(estimator=model)
+            output_meta = np.array([])
```
Comment (sarahyurick) [inline dask_sql/physical/rel/custom/create_model.py:187]: With this, `output_meta` is always `[]`

File: dask_sql/physical/rel/custom/predict.py [hunk 2]
```diff
@@ -59,7 +60,13 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
 
         model, training_columns = context.schema[schema_name].models[model_name]
         df = context.sql(sql_select)
-        prediction = model.predict(df[training_columns])
+        part = df[training_columns]
+        output_meta = model.predict_meta
+        if part.shape[0].compute() == 0 and output_meta is not None:
```
Comment (sarahyurick) [inline dask_sql/physical/rel/custom/predict.py:65]: `compute` is needed on the Delayed object. I believe that right now, `output_meta` will always be `[]`

File: dask_sql/physical/rel/custom/predict.py [hunk 3]
```diff
@@ -59,7 +60,13 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
 
         model, training_columns = context.schema[schema_name].models[model_name]
         df = context.sql(sql_select)
-        prediction = model.predict(df[training_columns])
+        part = df[training_columns]
+        output_meta = model.predict_meta
```
Comment (sarahyurick) [inline dask_sql/physical/rel/custom/predict.py:64]: AttributeError: 'KMeans' object has no attribute 'predict_meta'

File: dask_sql/physical/rel/custom/create_model.py (hunk 1 above)
Comment (VibhuJawa) [reply dask_sql/physical/rel/custom/create_model.py:187]: I dont think we can just hardcode the meta to be `output_meta` to be `np.array([])` .  We also use cuML for this case and that outputs a cuDF Series.

File: dask_sql/physical/rel/custom/predict.py (hunk 2 above)
Comment (VibhuJawa) [reply dask_sql/physical/rel/custom/predict.py:69]: You dont need to compute for this to  do this, we can do it lazily too.

File: dask_sql/physical/rel/custom/predict.py [hunk 4]
```diff
@@ -59,7 +61,16 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
...
+        part = df[training_columns]
+        try:
+            output_meta = model.predict_meta
+        except AttributeError:
+            output_meta = None
+        if part.shape[0].compute() == 0 and output_meta is not None:
+            empty_output = self.handle_empty_partitions(output_meta)
+            if empty_output is not None:
+                return empty_output
+        prediction = model.predict(part)
```
Comment (VibhuJawa) [inline dask_sql/physical/rel/custom/predict.py:73]: We should wrap the predict like the following for cases only for when we have a  `ParallelPostFit`  model.  


```python
  if isinstance(model, ParallelPostFit):
      output_meta = model.predict_meta
      if predict_meta is None:
         predict_meta = model.estimator.predict(part._meta_nonempty)
         
      prediction  = part.map_partitions(_predict, predict_meta, model.estimator, meta=predict_meta)

def _pedict(part, predict_meta, estimator):
    if part.shape[0] == 0 and predict_meta is not None:
        empty_output = handle_empty_partitions(output_meta)
        return empty_output
    return estimator.predict(part)
  ```

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
We should not hard code any meta values and should only handle cases when model is `ParallelPostFit ` .

**Review by VibhuJawa — CHANGES_REQUESTED** [review]
Please add more tests

**Pull request closed without merging**